src/TestFramework/Extension.WinUI/UITestMethodAttribute.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestTools.UnitTesting.AppContainer
+{
+    using global::System;
+    using global::System.Runtime.CompilerServices;
+
+    /// <summary>
+    /// Execute test code in UI thread for Windows store apps.
+    /// </summary>
+    public class UITestMethodAttribute : TestMethodAttribute
+    {
+        /// <summary>
+        /// Gets or sets the <see cref="Microsoft.System.DispatcherQueue"/> that should be used to invoke the UITestMethodAttribute.
+        /// If none is provided, it will try to use the Microsoft.UI.Xaml.Window.Current.DispatcherQueue, which only works on UWP.
+        /// </summary>
+        public static Microsoft.System.DispatcherQueue DispatcherQueue { get; set; }
```

Comment:
hmm, this would cause issues if the tests are run in parallel, wouldn't it?

Reply:
I can't see why it could, given that you provide one. The same concept was/is used in UWP, and it works fine (UWP used a static property to get the CoreDispatcher). Preferably, this should be set only once, and only if needed.